src/erls_resource.erl

```diff
@@ -80,6 +80,11 @@ do_request(#erls_params{host=Host, port=Port, timeout=Timeout, ctimeout=CTimeout
                 {ok, RespBody} -> {error, {Status, erls_json:decode(RespBody)}};
                 {error, _Reason} -> {error, Status}
             end;
+        {ok, ClientRef} ->
+            %% that's when the options passed to hackney included `async'
+            %% this reference can then be used to match the messages from
+            %% hackney when ES replies; see the hackney doc for more information
+            {ok, ClientRef};
         {error, R} ->
             {error, R}
```

Comment:
Note that I was tempted to replace the last 2 clauses, that essentially do nothing, by just a catch-all clause
`Else -> Else`, but that seemed to be somewhat opposed to the spirit of your code. Happy to change if you think it'd be better :) thanks!